From 7df8e53dc9cf89f6fbfd186efab15f5ad8a434b1 Mon Sep 17 00:00:00 2001
From: Paul Naoumenko <paulx.naoumenko@intel.com>
Date: Thu, 18 Apr 2013 18:32:00 +0200
Subject: [PORT FROM MAIN] wifi: Report error when failing to get netmask property

BZ: 94520

Exiting gracefully (with -1) from libnetutils
in case th netmask prop is not correcly retrieved.

Feature   : Wlan
Category  : Improvement
Purpose   : AOSP+
Origin    : Upstream
Upstream-candidate: yes

Change-Id: I545fc76e2c639e20637b16d270649497e1620a5b
Orig-Change-Id: I1af1ea6aada8febc8f895d28b332c5fff2cbf9b9
Signed-off-by: Paul Naoumenko <paulx.naoumenko@intel.com>
---
 libnetutils/dhcp_utils.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/libnetutils/dhcp_utils.c b/libnetutils/dhcp_utils.c
index bd48eff..de5165b 100644
--- a/libnetutils/dhcp_utils.c
+++ b/libnetutils/dhcp_utils.c
@@ -138,6 +138,9 @@ static int fill_ip_info(const char *interface,
             mask = mask << 1;
         }
         *prefixLength = p;
+    } else {
+        snprintf(errmsg, sizeof(errmsg), "Could not get the value of property:  %s", prop_name);
+        return -1;
     }
 
     for (x=0; dns[x] != NULL; x++) {
-- 
1.7.4.1

